internal/tmpl/character/tmpl.go

```diff
@@ -167,6 +167,25 @@ func (c *Tmpl) AddPreDamageMod(mod core.PreDamageMod) {
 	c.PreDamageMods[ind] = mod
 }
 
+func (c *Tmpl) DeletePreDamageMod(key string) {
+	ind := -1
+	for i, v := range c.PreDamageMods {
+		if v.Key == key {
+			ind = i
+		}
+	}
+	// return if mod does not exists
+	if ind == -1 {
+		return
+	}
+
+	// remove mod
+	c.PreDamageMods[ind].Event.SetEnded(c.Core.F)
+	c.PreDamageMods[ind] = c.PreDamageMods[len(c.PreDamageMods)-1]
```

Comment:
this would swap the last mod and the current mod right? wouldn't it be better to keep the ordering here?

something like this: https://github.com/genshinsim/gcsim/blob/460f90d8e46903b950587f0859b42910040b4ab0/internal/tmpl/construct/construct.go#L206

Reply:
yes though i don't think order matter for mods.. i guess it does for logs.